modules/subscription/variables.tf

```diff
@@ -196,3 +196,81 @@ variable "wait_for_subscription_before_subscription_operations" {
 The duration to wait after vending a subscription before performing subscription operations.
 DESCRIPTION
 }
+
+variable "subscription_budgets" {
+  type = map(object({
+    amount            = number
+    time_grain        = optional(string, "Monthly")
+    time_period_start = string
+    time_period_end   = string
+    notifications = map(object({
+      enabled       = bool
+      operator      = string # EqualTo, GreaterThan, GreaterThanOrEqualTo
+      threshold     = number # 0-1000 percent
+      thresholdType = string # Actual, Forecasted
```

Comment:
Question for the team - does it make sense to have this interface "look" like azurerm - e.g. the input name would be camel-cased, like "threshold_type", and the ones below like "contact_emails" etc.

My thoughts are if we do switch out the implementation for an AzureRM version, the interface will remain the same.

Reply:
Agreed, the inputs for all terraform modules should be snake case.

Reply:
added local to transform the notification map.